### PR TITLE
feat(dynamicWidgets): add default attributesToRender & transformItems

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

Retrieves the default attributesToRender from the result, and gives a default to transformItems now

**Result**

- [x] retrieve attributesToRender from the result
- [x] add a default for transformItems (identity)
- [x] update helper & client with new TS (https://github.com/algolia/algoliasearch-client-javascript/pull/1273, https://github.com/algolia/algoliasearch-helper-js/pull/823)

### CODE REVIEW

File: src/connectors/dynamic-widgets/__tests__/connectDynamicWidgets-test.ts [hunk 1]
```diff
@@ -39,17 +43,12 @@ describe('connectDynamicWidgets', () => {
       `);
     });
 
-    it('correct usage', () => {
+    it('minimal usage', () => {
```
Comment (francoischalifour) [inline src/connectors/dynamic-widgets/__tests__/connectDynamicWidgets-test.ts:46]: I don't think this test makes sense anymore if we don't take into account the first implem. It's like if we were testing that adding each option doesn't throw.

File: src/widgets/dynamic-widgets/__tests__/dynamic-widgets-test.ts [hunk 2]
```diff
@@ -51,21 +51,16 @@ describe('dynamicWidgets()', () => {
       `);
     });
 
-    test('transformItems is required', () => {
+    test('transformItems is not required', () => {
```
Comment (francoischalifour) [inline src/widgets/dynamic-widgets/__tests__/dynamic-widgets-test.ts:54]: Same for that test.

**Review by francoischalifour — APPROVED** [review]

File: src/widgets/dynamic-widgets/__tests__/dynamic-widgets-test.ts (hunk 2 above)
Comment (Haroenv) [reply src/widgets/dynamic-widgets/__tests__/dynamic-widgets-test.ts:54]: good point, not needed anymore for both

**Review by eunjae-lee — APPROVED** [review]
Looks great without the temporary code any more!

**Pull request merged**